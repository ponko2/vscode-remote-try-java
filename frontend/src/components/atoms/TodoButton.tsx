import classnames from "classnames";
import React from "react";
import styles from "./TodoButton.module.css";

type Props = React.DetailedHTMLProps<
  React.ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>;

export const TodoButton = ({
  className,
  children,
  ...restProps
}: Props): JSX.Element => (
  <button className={classnames([styles.button, className])} {...restProps}>
    {children}
  </button>
);
