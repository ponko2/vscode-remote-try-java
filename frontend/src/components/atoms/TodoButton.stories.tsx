import { action } from "@storybook/addon-actions";
import type { ComponentMeta, ComponentStory } from "@storybook/react";
import { TodoButton } from "./TodoButton";

export default {
  component: TodoButton,
} as ComponentMeta<typeof TodoButton>;

const Template: ComponentStory<typeof TodoButton> = () => (
  <TodoButton onClick={action("clicked")}>Hello, World!!</TodoButton>
);

export const Basic = Template.bind({});
