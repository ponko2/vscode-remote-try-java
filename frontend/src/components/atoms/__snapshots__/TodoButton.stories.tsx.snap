// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/atoms/TodoButton Basic smoke-test 1`] = `
<button class="_button_z78gq_1">
  Hello, World!!
</button>
`;
